pam/pam.go

```diff
@@ -354,12 +354,15 @@ func (h *pamModule) AcctMgmt(mTx pam.ModuleTransaction, flags pam.Flags, args []
 
 // newClient returns a new GRPC client ready to emit requests.
 func newClient(args map[string]string) (client authd.PAMClient, close func(), err error) {
-	conn, err := grpc.Dial("unix://"+getSocketPath(args), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	dialCtx, dialCancel := context.WithTimeout(context.TODO(), time.Second*5)
```

Comment:
here, you are adding 5 seconds while one NSS, you did 10, as said the other day, I would prefer 5, but at least, we should keep the same for both, shouldn’t we?

Reply:
I'm not sure I agree 100%, they are different services after all. I used 5 seconds here because it was the same duration used in the timeout right below (to wait for the daemon to be ready).

Reply:
In pam module we're also doing 5 seconds, so imho it would be nice to have a common constant that is shared everywhere.

While... Ideally I think we should allow customizing this from settings too (at least on module parameters is easy to do).

Reply:
I'm fine with toning down the NSS timeout to 5 seconds. The PR page is very busy today, would you mind if I added the change in this PR instead?

Reply:
I don’t mind if you add it to that PR if you check the error code of NSS too and return the appropriate one as we did when we couldn’t open the database in aad-auth :)

Reply:
Yeah, in NSS we kept the "same" behaviour, i.e. return `Unavail` if we can't connect to authd (look [here](https://github.com/ubuntu/authd/blob/c7efeef36080e00f7774b67c985009bb04119039/nss/src/lib.rs#L46) and [here](https://github.com/ubuntu/authd/blob/c7efeef36080e00f7774b67c985009bb04119039/nss/src/passwd/mod.rs#L71)).